src/platform/forms-system/src/js/validation.js

```diff
@@ -405,6 +405,33 @@ export function validateCurrentOrPastDate(
   }
 }
 
+export function validateCurrentOrPastMemorableDate(
+  errors,
+  dateString,
+  formData,
+  schema,
+  errorMessages = {},
+) {
+  const {
+    futureDate = 'Please provide a valid current or past date',
+  } = errorMessages;
+  validateDate(
+    errors,
+    dateString,
+    formData,
+    schema,
+    errorMessages,
+    undefined,
+    undefined,
+    minYear,
+    new Date().getFullYear(),
+  );
+  const { day, month, year } = parseISODate(dateString);
+  if (!day || !year || !day || !isValidCurrentOrPastDate(day, month, year)) {
```

Comment:
this block is copied from the validation directly above. but I added this part for checking the day or year or month.